pkgs/applications/networking/browsers/microsoft-edge/update.py

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i python3 -p python3Packages.packaging python3Packages.debian
+
+import base64
+import gzip
+import textwrap
+from urllib import request
+
+from debian.deb822 import Packages
+from debian.debian_support import Version
+
+
+def packages():
+    packages_url = 'https://packages.microsoft.com/repos/edge/dists/stable/main/binary-amd64/Packages.gz'
+    handle = request.urlopen(packages_url)
+    data = gzip.GzipFile(fileobj=handle).read()
+    return data
+
+
+def latest_packages(packages: bytes):
+    latest_packages: dict[str, Packages] = {}
+    for package in Packages.iter_paragraphs(packages, use_apt_pkg=False):
+        name: str = package['Package']
+        if not name.startswith('microsoft-edge-'):
+            continue
+        channel = name.replace('microsoft-edge-', '')
+        if channel not in latest_packages:
+            latest_packages[channel] = package
+        else:
+            old_package = latest_packages[channel]
+            if old_package.get_version() < package.get_version():  # type: ignore
+                latest_packages[channel] = package
+    return latest_packages
+
+
+def nix_expressions(latest: dict[str, Packages]):
+    channel_strs: list[str] = []
+    for channel, package in latest.items():
+        print(f"Processing {channel} {package['Version']}")
+        match = Version.re_valid_version.match(package['Version'])
+        assert match is not None
+
+        version = match.group('upstream_version')
+        revision = match.group('debian_revision')
+        sri = 'sha256-' + \
+            base64.b64encode(bytes.fromhex(package['SHA256'])).decode('ascii')
+
+        channel_str = textwrap.dedent(
+            f'''\
+            {channel} = import ./browser.nix {{
+              channel = "{channel}";
+              version = "{version}";
+              revision = "{revision}";
+              sha256 = "{sri}";
+            }};'''
+        )
+        channel_strs.append(channel_str)
+    return channel_strs
+
+
+def write_expression():
+    latest = latest_packages(packages())
+    channel_strs = nix_expressions(latest)
+    nix_expr = '{\n' + textwrap.indent('\n'.join(channel_strs), '  ') + '\n}'
+
+    print(textwrap.dedent(
+        '''
+        This script will update the default.nix file with the following contents:
+        {}\n
+        Are you sure you want to continue [y/N]?
+        '''
+    ).format(nix_expr))
```

Comment:
Please don't display any annoying prompts, it is fine to just overwrite the file since we are git version controlled.

Reply:
The prompt is suggested by @rikhuijzer. 

If overwriting the desired behavior, I also think it's fine to update default.nix silently.

By the way, is it possible to make the update automatically using metadata like `passthru.updateScript`?

Reply:
> Of course, that is the intended way to use update scripts and why prompts are normally not done.

However, this `default.nix` creates three different packages. I'm not sure setting the update script for all these packages will cause problems or not.